src/clusterfuzz/_internal/google_cloud_utils/credentials.py

```diff
@@ -58,22 +62,45 @@ def get_default(scopes=None):
   """Get default Google Cloud credentials."""
   if _use_anonymous_credentials():
     return credentials.AnonymousCredentials(), ''
-
   return google.auth.default(scopes=scopes)
 
 
-def get_signing_credentials():
-  """Returns signing credentials for signing URLs."""
+def _set_gcs_signing_service_account():
+  project_id = utils.get_application_id()
+  service_account_key = secret_manager.get(_SIGNING_KEY_SECRET_ID, project_id)
+  service_account_key_path = os.path.join(
+      environment.get_value('ROOT_DIR'), 'bot', 'gcs_key')
+  with open(service_account_key_path, 'w') as fp:
+    fp.write(service_account_key)
+  return service_account_key_path
+
+
+@retry.wrap(
+    retries=FAIL_RETRIES,
+    delay=FAIL_WAIT,
+    function='google_cloud_utils.credentials.get_signing_service_account')
+def get_storage_signing_service_account():
+  """Gets a dedicated signing account for signing storage objects."""
   if _use_anonymous_credentials():
     return None
-
   google_application_credentials = os.getenv('GOOGLE_APPLICATION_CREDENTIALS',
                                              None)
-  if google_application_credentials is not None:
+  if google_application_credentials:
+    return google_application_credentials
+
+  return _set_gcs_signing_service_account()
+
+
+def get_signing_credentials(service_account_path):
+  """Returns signing credentials for signing URLs."""
+  if _use_anonymous_credentials():
+    return None
+
+  if service_account_path is not None:
     # Handle cases like android and Mac where bots are run outside of Google
     # Cloud Platform and don't have access to metadata server.
     signing_creds = service_account.Credentials.from_service_account_file(
-        google_application_credentials, scopes=_SCOPES)
+        service_account, scopes=_SCOPES)
```

Comment:
Does it work if we pass the JSON decoded service account dict to https://google-auth.readthedocs.io/en/master/reference/google.oauth2.service_account.html#google.oauth2.service_account.Credentials.from_service_account_info ? 

This way we avoid the complexity of writing out to a file and keep everything in memory.

Reply:
I'll check tomorrow (also when i'll write tests).

Reply:
I think it should work given the example here: https://google-auth.readthedocs.io/en/master/reference/google.oauth2.service_account.html#domain-wide-delegation

```
service_account_info = json.load(open('service_account.json'))
credentials = service_account.Credentials.from_service_account_info(
    service_account_info)
```

Reply:
It will add some complexity because we are already using the file based version on mac/linux/android. Maybe I'll just stop supporting GOOGLE_APPLICATION_CREDENTIALS for signing entirely?

Reply:
Yes let's keep things homogeneous and use this account everywhere. The existing service accounts should have access to these secrets.